File: Atividade-2/javascript/ExecutadorArquivo.js
const fs = require('fs');

class ExecutadorArquivo {
    executa(filePath, listaLigada) {
        const buffRead = fs.readFileSync(filePath, 'utf8').split('\n');

        const numerosIniciais = buffRead[0].split(' ');
        for (const numero of numerosIniciais) {
            listaLigada.insereFim(parseInt(numero));
        }

        for (let i = 2; i < buffRead.length; i++) {
            const linha = buffRead[i].trim();
            if (linha) {
                console.log(`Ação ${i - 1}:`);
                const acaoSplit = linha.split(' ');

                if (acaoSplit[0] === 'A') {
                    const valor = parseInt(acaoSplit[1]);
                    const posicao = acaoSplit[2] ? parseInt(acaoSplit[2]) - 1 : null;

                    if (posicao !== null) {
                        try {
                            listaLigada.insereElementoPosicao(valor, posicao);
                            console.log(`Valor ${valor} inserido na posição ${posicao + 1}.`);
                        } catch (e) {
                            console.log('Posição para inserir inválida. Continuando para a próxima ação.');
                        }
                    }
                    else {
                        console.log('Posição para inserir inválida. Continuando para a próxima ação.');
                        continue;
                    }
                } else if (acaoSplit[0] === 'R') {
                    const valor = parseInt(acaoSplit[1]);

                    //quando a ação remover é passado a posição
                    if (acaoSplit.length == 3) {
                        
                        const posicao = parseInt(acaoSplit[2]) - 1;
                        const valorPosicao = listaLigada.buscaIndice(posicao);
                        const valorExiste = listaLigada.buscaElemento(valor);

                        if (valorExiste && (valorPosicao == valor)) {
                            listaLigada.remove(valor);
                            console.log("Valor " + valor + " removido.");
                        } else {
                            console.log("O valor " + valor + " não existe na posição " + (posicao + 1) + ". Continuando para a próxima ação.");
                        }
                    }
                    //quando a ação remover é passado somente o valor
                    else if (listaLigada.buscaElemento(valor)) {
                        listaLigada.remove(valor);
                        console.log("Valor " + valor + " removido.");
                    } else {
                        console.log("O valor " + valor + " não existe na lista. Continuando para a próxima ação.");
                    }

                } else if (acaoSplit[0] === 'P') {
                    console.log(listaLigada.toString());
                    console.log('-------');
                }
            }
        }
    }
}

module.exports = ExecutadorArquivo;

File: Atividade-2/python/ExecutadorArquivo.py
import io

class ExecutadorArquivo:

    def executa(filePath, listaLigada):
        with io.open(filePath, 'r', encoding='utf8') as file:
            linha = file.readline().strip()

            numerosIniciais = linha.split(" ")
            
            for numero in numerosIniciais:
                listaLigada.insereFim(int(numero))

            quantidadeAcoes = int(file.readline().strip())

            for i in range(quantidadeAcoes):
                print("Ação " + str(i + 1) + ":")
                acao = file.readline().strip()
                
                acaoSplit = acao.split(" ")

                if acaoSplit[0] == "A":
                    
                    valor = int(acaoSplit[1])

                    try:
                        posicao = int(acaoSplit[2]) - 1
                        listaLigada.insereElementoPosicao(valor, posicao)
                        print("Valor " + str(valor) + " inserido na posição " + str(posicao + 1) + ".")
                    except Exception as e:
                        print("Posição para inserir inválida. Continuando para a próxima ação.")

                elif acaoSplit[0] == "R":

                    valor = int(acaoSplit[1])
                    
                    # quando a ação remover é passado a posição
                    if len(acaoSplit) == 3:
                        
                        posicao = int(acaoSplit[2]) - 1
                        valorPosicao = listaLigada.buscaIndice(posicao)
                        valorExiste = listaLigada.buscaElemento(valor)

                        if valorExiste and (valorPosicao == valor):
                            listaLigada.remove(valor)
                            print("Valor " + str(valor) + " removido.")
                        else:
                            print("O valor " + str(valor) + " não existe na posição "+ str(posicao+1) +". Continuando para a próxima ação.")
                    # quando a ação remover é passado somente o valor
                    elif listaLigada.buscaElemento(valor):
                        listaLigada.remove(valor)
                        print("Valor " + str(valor) + " removido.")
                    else:
                        print("O valor " + str(valor) + " não existe na lista. Continuando para a próxima ação.")
                
                elif acaoSplit[0] == "P":
                    
                    print(str(listaLigada))
                    print("-------")